Content.Server/Morgue/MorgueSystem.cs

```diff
@@ -1,143 +1,281 @@
 using Content.Server.Morgue.Components;
 using Content.Shared.Morgue;
 using Content.Shared.Examine;
-using Content.Shared.Database;
-using Content.Shared.Verbs;
-using JetBrains.Annotations;
-using Content.Shared.Interaction.Events;
 using Robust.Server.GameObjects;
-using Content.Server.Players;
-using Content.Server.GameTicking;
 using Content.Server.Popups;
 using Content.Shared.Standing;
 using Robust.Shared.Player;
+using Robust.Shared.Audio;
+using Content.Server.Storage.Components;
+using Content.Server.Storage.EntitySystems;
+using Content.Shared.Body.Components;
+using Robust.Server.Containers;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Content.Shared.Interaction;
+using Content.Shared.Verbs;
+using Content.Server.GameTicking;
+using Content.Shared.Interaction.Events;
 
-namespace Content.Server.Morgue
+namespace Content.Server.Morgue;
+
+/// <summary>
+///     This is the system for morgues but is also used for 
+///     crematoriums. Anything with a slab that you stick
+///     bodies into would work as well.
+/// </summary>
+public sealed partial class MorgueSystem : EntitySystem
 {
-    [UsedImplicitly]
-    public sealed class MorgueSystem : EntitySystem
+    [Dependency] private readonly GameTicker _ticker = default!;
+    [Dependency] private readonly EntityStorageSystem _entityStorage = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
+    [Dependency] private readonly StandingStateSystem _standing = default!;
+    [Dependency] private readonly ContainerSystem _container = default!;
+
+    public override void Initialize()
     {
-        [Dependency] private readonly GameTicker _ticker = default!;
-        [Dependency] private readonly PopupSystem _popup = default!;
-        [Dependency] private readonly StandingStateSystem _stando = default!;
+        base.Initialize();
+
+        SubscribeLocalEvent<MorgueComponent, ComponentInit>(OnInit);
 
-        private float _accumulatedFrameTime;
+        SubscribeLocalEvent<MorgueComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<MorgueComponent, GetVerbsEvent<AlternativeVerb>>(AddCremateVerb);
+        SubscribeLocalEvent<MorgueComponent, SuicideEvent>(OnSuicide);
+        SubscribeLocalEvent<MorgueTrayComponent, StorageBeforeCloseEvent>(OnStorageBeforeClose);
+
+        //These are used to intercept and override the entityStorage opening functionality.
+        SubscribeLocalEvent<MorgueComponent, ActivateInWorldEvent>(ToggleMorgueSlab,
+            before: new[] { typeof(EntityStorageSystem) });
```

Comment:
Its fine for now, but eventually the tray should probably just be removed altogether, and the morgue component should just be something that handles beeping & visuals that trigger on storage open & close events.